cfme/services/catalogs/service_catalogs.py

```diff
@@ -33,47 +39,74 @@
-                    lambda ctx: accordion_tree(
-                        'All Services', ctx['catalog'], ctx['catalog_item'].name),
-                    {
-                        'order_service_catalog': lambda _: sel.click(order_button)
-                    }
-                ]
-            }
-        ]
-    }
-)
-
-
-class ServiceCatalogs(Updateable, Pretty):
+match_page = partial(match_location, title='Catalogs', controller='catalog')
+
+
+def nav_to_all():
+    from cfme.web_ui.menu import nav
+    nav._nav_to_fn('Services', 'Catalogs')(None)
+    tree = accordion.tree('Service Catalogs')
+    tree.click_path('All Services')
+
+
+class ServiceCatalogs(Updateable, Pretty, Navigatable):
     pretty_attrs = ['service_name']
 
-    def __init__(self, service_name=None, stack_data=None):
+    def __init__(self, service_name=None, stack_data=None, appliance=None):
         self.service_name = service_name
         self.stack_data = stack_data
+        Navigatable.__init__(self, appliance=appliance)
 
-    def order(self, catalog, catalog_item):
-        sel.force_navigate('order_service_catalog',
-                           context={'catalog': catalog,
-                                    'catalog_item': catalog_item})
+    def order(self):
+        navigate_to(self, 'Order')
         sel.click(form_buttons.submit)
         wait_for(flash.get_messages, num_sec=10, delay=2)
         flash.assert_success_message("Order Request was Submitted")
 
-    def order_stack_item(self, catalog, catalog_item):
-        sel.force_navigate('order_service_catalog',
-                           context={'catalog': catalog,
-                                    'catalog_item': catalog_item})
+    def order_stack_item(self):
+        navigate_to(self, 'Order')
         stack_form.fill(self.stack_data)
         sel.click(form_buttons.submit)
         wait_for(flash.get_messages, num_sec=10, delay=2)
         flash.assert_success_message("Order Request was Submitted")
+
+
+@navigator.register(ServiceCatalogs, 'All')
+class ServiceCatalogsAll(CFMENavigateStep):
+    prerequisite = NavigateToAttribute('appliance.server', 'LoggedIn')
+
+    def am_i_here(self):
+        return match_page(summary='All Services')
+
+    def step(self):
+        nav_to_all()
+
+    def resetter(self):
+        tb.refresh()
+        tb.select('List View')
+        # Ensure no rows are checked
+        if paginator.page_controls_exist():
+            sel.check(paginator.check_all())
+            sel.uncheck(paginator.check_all())
+
+
+@navigator.register(ServiceCatalogs, 'Details')
+class ServiceCatalogDetails(CFMENavigateStep):
+    prerequisite = NavigateToSibling('All')
+
+    def step(self):
+        listview_table.click_row_by_cells({'Name': self.obj.service_name})
+
+    def resetter(self):
+        tb.refresh()
+
+
+@navigator.register(ServiceCatalogs, 'Order')
+class ServiceCatalogOrder(CFMENavigateStep):
+    prerequisite = NavigateToSibling('Details')
+
+    def am_i_here(self):
```

Comment:
I'm testing localling and seeing other small problems (that I duplicated in my example above) like `self.service_name` instead of `self.obj.service_name`